boolean-search/booleansearch.app.js

```diff
@@ -78,25 +78,27 @@ var setSourceForFilter;
         })
         .value('headerInfo', {
             pid: "",
-            cid: ""
+            cid: "",
+            wid: ""
         })
-        .factory('filterOptions', ['$http', '$window', 'ERMrest', 'headInjector', 'MathUtils', 'UriUtils', 'headerInfo', function ($http, $window, ERMrest, headInjector, MathUtils, UriUtils, headerInfo) {
+        .factory('filterOptions', ['$http', '$window', 'ERMrest', 'headInjector', 'MathUtils', 'UriUtils', 'headerInfo','ConfigUtils', function ($http, $window, ERMrest, headInjector, MathUtils, UriUtils, headerInfo,ConfigUtils) {
             var baseUrl = $window.location.origin;
             var specExprUrl = baseUrl + "/ermrest/catalog/2/attributegroup/Gene_Expression:Specimen_Expression";
             var devStageUrl = baseUrl + "/ermrest/catalog/2/attribute/Vocabulary:Developmental_Stage";
             var sourceUrl = baseUrl + "/ermrest/catalog/2/entity/Vocabulary:Anatomy";
 
             // Configuring ERMrestjs service object and http module
-            var contextHeaderParams = { "cid": "boolean-search" };
+            var contextHeaderParams = ConfigUtils.getContextHeaderParams();
             var server = ERMrest.ermrestFactory.getServer(baseUrl + "/ermrest", contextHeaderParams);
-            headerInfo.pid = MathUtils.uuid();
-            headerInfo.cid = "boolean-search";
-            headInjector.setWindowName();
+            headerInfo.pid = contextHeaderParams.pid;
+            headerInfo.cid = contextHeaderParams.cid;
+            headerInfo.wid = contextHeaderParams.wid;
             var getHeader = function () {
                 return {
-                    wid: $window.name,
+                    wid: headerInfo.wid,
                     cid: headerInfo.cid,
                     pid: headerInfo.pid,
+                    catalog:UriUtils.getCatalogId(),
```

Comment:
I don't think that catalog id will be set at this point since it won't be in the url. We are going to have to hardcode the catalog id for this case.

Reply:
Updated in the PR to hardcode catalog_id as "2"